llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -517,25 +517,31 @@ static bool isUnpackedVectorVT(EVT VecVT) {
 static InstructionCost getHistogramCost(const IntrinsicCostAttributes &ICA) {
   Type *BucketPtrsTy = ICA.getArgTypes()[0]; // Type of vector of pointers
   Type *EltTy = ICA.getArgTypes()[1];        // Type of bucket elements
+  unsigned TotalHistCnts = 1;
 
-  // Only allow (32b and 64b) integers or pointers for now...
+  // Only allow (up to 64b) integers or pointers
```

Comment:
I'd suggest adding a variable `unsigned EltSize = EltTy->getScalarSizeInBits();` and using it both in the initial size check and the total size check below.

Reply:
Done.